sklearn/cluster/hierarchical.py

```diff
@@ -713,7 +713,7 @@ def fit(self, X, y=None):
         -------
         self
         """
-        X = check_array(X)
+        X = check_array(X, ensure_min_samples=2)
```

Comment:
The problem is that it is really hard to test that this PR doesn't break anything. I thought it was a good idea to include a test that sensible things happen with 2d X with one sample or one feature. It turns out that many of our estimators break.
This should probably be a different PR though.